commerce-notification-test/src/testIntegration/java/com/liferay/commerce/notification/test/CommerceNotificationTest.java

```diff
@@ -0,0 +1,368 @@
+
+		_commerceAccount = CommerceAccountTestUtil.addBusinessCommerceAccount(
+			_accountAdmin.getUserId(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			new long[] {_user.getUserId()}, null, _serviceContext);
+
+		_accountAdminRole = _roleLocalService.fetchRole(
+			_company.getCompanyId(),
+			CommerceAccountConstants.ACCOUNT_ADMINISTRATOR_ROLE_NAME);
+
+		if (_accountAdminRole == null) {
+			_accountAdminRole = _roleLocalService.addRole(
+				_serviceContext.getUserId(), null, 0,
+				CommerceAccountConstants.ACCOUNT_ADMINISTRATOR_ROLE_NAME,
+				Collections.singletonMap(
+					_serviceContext.getLocale(),
+					CommerceAccountConstants.ACCOUNT_ADMINISTRATOR_ROLE_NAME),
+				Collections.emptyMap(), RoleConstants.TYPE_SITE,
+				StringPool.BLANK, _serviceContext);
+
+			_createdAdminRole = true;
+		}
+
+		_userGroupRoleLocalService.deleteUserGroupRoles(
+			_accountAdmin.getUserId(),
+			new long[] {_commerceAccount.getCommerceAccountGroupId()});
+
+		_userGroupRoleLocalService.addUserGroupRoles(
+			_accountAdmin.getUserId(),
+			_commerceAccount.getCommerceAccountGroupId(),
+			new long[] {_accountAdminRole.getRoleId()});
+	}
+
+	private CommerceNotificationTemplate _setUpNotificationTemplate(
+			String to, String notificationType)
+		throws PortalException {
+
+		return CommerceNotificationTestUtil.addCommerceNotificationTemplate(
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), to, notificationType,
+			_serviceContext);
+	}
+
+	private void _setUpOrderManager() throws Exception {
+		_orderManager = UserTestUtil.addUser(
+			_user.getCompanyId(), _user.getUserId(), "businessUser",
+			_serviceContext.getLocale(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(),
+			new long[] {_serviceContext.getScopeGroupId()}, _serviceContext);
+
+		_orderManagerRole = _roleLocalService.fetchRole(
+			_company.getCompanyId(), "Order Manager");
+
+		if (_orderManagerRole == null) {
+			_orderManagerRole = _roleLocalService.addRole(
+				_serviceContext.getUserId(), null, 0, "Order Manager",
+				Collections.singletonMap(
+					_serviceContext.getLocale(), "Order Manager"),
+				Collections.emptyMap(), RoleConstants.TYPE_SITE,
+				StringPool.BLANK, _serviceContext);
+
+			_createdOrderManagerRole = true;
+		}
+
+		_userGroupRoleLocalService.deleteUserGroupRoles(
+			_orderManager.getUserId(),
+			new long[] {_commerceAccount.getCommerceAccountGroupId()});
+
+		_userGroupRoleLocalService.addUserGroupRoles(
+			_orderManager.getUserId(),
+			_commerceAccount.getCommerceAccountGroupId(),
+			new long[] {_orderManagerRole.getRoleId()});
+	}
+
+	@DeleteAfterTestRun
+	private User _accountAdmin;
+
+	private Role _accountAdminRole;
+	private CommerceAccount _commerceAccount;
+
+	@Inject
+	private CommerceChannelLocalService _commerceChannelLocalService;
+
+	private CommerceCurrency _commerceCurrency;
```

Comment:
I really think we should be cleaning up these objects after the test.... I would guess that `@DeleteAfterTestRun`'ing the `Company` calls `companyLocalServiceImpl.doDeleteCompany`, which deletes a lot of portal content specific to the company, but won't delete any of our custom data added against that `companyId`. Not a huge deal since the same `companyId` _shouldn't_ be reused, but it's possible this _could_ lead to flaky tests in the future, and it will probably add stale data to the db of any dev who runs these tests.